lib/cmsis_rtos_v1/cmsis_thread.c

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel_structs.h>
+#include <atomic.h>
+#include <cmsis_os.h>
+
+static s32_t zephyr_to_cmsis_priority(u32_t z_prio)
+{
+	return(osPriorityRealtime - z_prio);
+}
+
+static u32_t cmsis_to_zephyr_priority(s32_t c_prio)
+{
+	return(osPriorityRealtime - c_prio);
+}
+
+static void zephyr_thread_wrapper(void *arg1, void *arg2, void *arg3)
+{
+	void * (*fun_ptr)(void *) = arg3;
+
+	fun_ptr(arg1);
```

Comment:
Why even bother with this wrapper function?  ARM arguments are passed in registers, so there's no reason not to just stuff the fun_ptr in arg3 into the k_thread_create() call and save a whole stack frame.

Reply:
Well.. this is mainly because the 2 entrypoint function pointers are incompatible.  Zephyr is void (*)(void *, void *, void *) where as cmsis is void (* const)(const void *). I see it is mostly safe here but am just paranoid and would like to keep it the way it is.